Task3/src/main/java/Building.java

```diff
@@ -0,0 +1,108 @@
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class Building {
```

Comment:
We have to split this class. After spliting we should have:
1) Runner, which just use logic from other classes in order to run our tasks. 
2) PropertyUtil class which reads properties 
3) Building class which provides preparation logic for starting our tasks.
To summarize, after the changes, we should have a building class that accepts properties, a util property that provides properties, and a runner as a facade that connects the logic of both of these classes and runs our tasks.